media/sourcebuffer-changetype.js

```diff
@@ -0,0 +1,50 @@
+const mediaSource = new MediaSource();
+const audio = document.querySelector('audio');
+audio.src = URL.createObjectURL(mediaSource);
+
+mediaSource.addEventListener('sourceopen', function() {
+  URL.revokeObjectURL(audio.src);
+  feedAudioElement(0 /* index */);
+});
+
+const resources = [
+  { url: 'bear-opus.webm',     type: 'audio/webm; codecs="opus"' },
+  { url: 'bear-flac_frag.mp4', type: 'audio/mp4; codecs="flac"' }
+];
+
+async function feedAudioElement(index) {
+  if (index == resources.length) {
+    mediaSource.endOfStream();
+    return;
+  }
+
+  try {
+    let sourceBuffer;
+    const resource = resources[index];
+
+    if (mediaSource.sourceBuffers.length == 0) {
+      log('Adding source buffer with type: ' + resource.type);
+      sourceBuffer = mediaSource.addSourceBuffer(resource.type);
+      sourceBuffer.mode = 'sequence';
```

Comment:
Perhaps add comment that 'sequence' is used here purely for simplicity, and that a more sophisticated MSE app would know which timestampOffset and appendWindow to use for each appendBuffer operation when splicing media.